cmd/influx_inspect/reportdisk/reportdisk.go

```diff
@@ -0,0 +1,308 @@
+// Package report reports statistics about TSM files.
+package reportdisk
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/influxdata/influxdb/models"
+	"github.com/influxdata/influxdb/pkg/reporthelper"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/influxdb/tsdb/engine/tsm1"
+)
+
+// Command represents the program execution for "influxd report".
+type Command struct {
+	Stderr io.Writer
+	Stdout io.Writer
+
+	dir      string
+	pattern  string
+	detailed bool
+}
+
+// NewCommand returns a new instance of Command.
+func NewCommand() *Command {
+	return &Command{
+		Stderr: os.Stderr,
+		Stdout: os.Stdout,
+	}
+}
+
+// Run executes the command.
+func (cmd *Command) Run(args ...string) error {
+	fs := flag.NewFlagSet("report", flag.ExitOnError)
+	fs.StringVar(&cmd.pattern, "pattern", "", "Include only files matching a pattern")
+	fs.BoolVar(&cmd.detailed, "detailed", false, "Report disk size by measurement")
```

Comment:
I wonder if a start & end time range would be useful as well. E.g., in some cases, a user may have months or years worth of data but notices a sudden spike in disk usage over the past week. Narrowing the time range would make the command run faster, use less resources, and still find the offending measurement(s).

Reply:
It is possible to run the command on a specific shard directory, which would narrow down the time - would that satisfy the need you are talking about?

Reply:
the 'pattern' argument should also help there.

Reply:
ah, right...running on a single shard directory is fine